### DISCUSSION TITLE
My ideas

### DISCUSSION THREAD

**Discussion by Gorilli09** [SuperTux/supertux#2636] [Ideas]
Worlds
1. The North
2. Desert
3. Small
4. Big
5. Jungle
6. Cloud
7. Lava
8. Space

Steam
1. Steam Trading Cards
2. Steam Workshop support
3. Steam Deck support

Controller
1. Hide mouse cursor when using controller
2. Controller vibration
3. Switchable controller buttons icons: Xbox 360 Controller, DualShock 3, Xbox Wireless Controller, DualShock 4, DualSense, Nintendo Switch Pro Controller, Steam Controller, Steam Deck, and Vanilla Generic Controller. I recommend Xelu's Free Controller Icons.
4. Peek with Right Stick

Editor
1. Controller support for Steam Deck users
2. MP3 and FLAC file support
3. Custom texture support
4. SDK
5. Custom content in Documents folder (this would make it easier to access and be cloud-saved through OneDrive)

Other
1. Cutscenes don't time-consume
2. Power-up Shop
3. Achievements
4. Entering pipes
5. Enemy encounters in world map (similar to Hammer Bros. in Super Mario Bros. 3)
6. Switching level planes (switching from foreground to background)
7. Lava instantly kills
8. Time attack
9. Online leaderboards
10. Mute on focus lost
11. Secondary bindings
12. Razer Chroma support

**Comment by provigz** [comment]
@Frostwithasideofsalt @tylerandari13 No spamming please.

**Comment by provigz** [comment]
Regarding the worlds, the main story mode worlds are already planned as the following:

1. Icy Island
2. Rooted and Corrupted Forest
3. Tropical Paradise
4. Mountain Peak

Regarding some the others, which I could comment on:

**Controller:**

1. I agree, this would also be good for Android builds too.

**Editor:**

3. What do you mean by "custom tilemap"?
5. Custom content should be available in "Appdata/Roaming" on Windows, since that is a standard. I believe home directory placement on Linux is also a standard. Essentially, in my opinion, using the `prefdir` PhysicsFS provides, as we currently do, is to be preferred.

**Other:**

1. Is available in Nightly builds, will be applied to new levels.
2. Power-up shop has been an idea for a while now, currently it still seems to be in discussion regarding if it should exist, and in what form exactly.
3. Achievements are planned to come sometime in the future.
7. I don't agree lava should instantly kill, the current way it works achieves consistency with the general power-up mechanic. Plus, lava not killing instantly can be used for interesting level design concepts, I believe.
8. In my opinion this doesn't exactly suit SuperTux, since it's not a speedrunning-oriented game. Alhough SuperTux is a choice for a game to speedrun for many people, and is very fun to speedrun, in my opinion, this doesn't seem to be a part of the game's main focus.
